src/genie/libs/parser/iosxr/show_tacacs.py

```diff
@@ -0,0 +1,146 @@
+"""show_tacacs.py
+   IOSXR parsers for the following show commands:
+     *  show tacacs
+"""
+# python
+
+# Metaparser
+# python
+import re
+
+# Metaparser
+from genie.metaparser import MetaParser
+from genie.metaparser.util.schemaengine import Any, Optional
+
+
+# ==================================================
+# Schema for 'show tacacs'
+# ==================================================
+class ShowTacacsSchema(MetaParser):
+    """Schema for show tacacs"""
+
+    schema = {
+        'server': {
+            Any(): {
+                Optional('vrf'): str,
+                Optional('server_type'): str,
+                'opens': str,
+                'closes': str,
+                'aborts': str,
+                'errors': str,
+                'packets_in': str,
+                'packets_out': str,
+                'status': str,
+                'single_connect': str,
+                'family': str,
+            },
+        },
+    }
+
+
+# ==================================================
+# Parser for 'show tacacs'
+# ==================================================
+class ShowTacacs(ShowTacacsSchema):
+    """Parser for show tacacs"""
+
+    cli_command = ['show tacacs']
+
+    def cli(self, output=None):
+        if output is None:
+            cmd = self.cli_command[0]
+            out = self.device.execute(cmd)
+        else:
+            out = output
+
+        # initial dictionary
+        resultdict = dict()
+        serverdict = dict()
+
+        # RegEX for first line with vrf
+        # Server: 127.0.0.1/24 vrf=default [private]
+        p1 = re.compile(
+            r'Server\:\s(?P<ip>[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\/[0-9]+)\svrf=(?P<vrf>[A-Za-z0-9]+)\s?\[(?P<server_type>\w+)\]')
```

Comment:
Looks like these regex statements could be improved to make them more readable/flexible if this parser needs to be expanded in the future.

For example it's best to have them start and end with ^ and $ respectively, so they don't start or end reading anywhere but the beginning/end of the line.

In addition it's also best to use things like \d, \S, or \w in lieu of [0-9] or [a-zA-Z].